Reactive Architecture

Kavignon opened the pull request:
Giving the final touch to the reactive architecture !

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/gameoflightandshadows/a-game-of-light-and-shadows/106)
<!-- Reviewable:end -->

Commit:
#88 implemented basic version of the team party manager. The rest of the protocol needs the dynamic prototype to be completed before matching on those cases,.

Commit:
#82
The item store can receive the information from the team party manager.
The store can validate that the good mode was selected before doing any transaction !

Commit:
#82 -
Added getter and setters functions implemented as lens to generically pair them as a tuple when getting and equipment new equipment

review-ninja commented:
<a href="https://app.review.ninja/GameOfLightAndShadows/A-game-of-light-and-shadows/pull/106" target="_blank"><img src="https://app.review.ninja/66727011/pull/106/badge" alt="ReviewNinja"/>

Commit:
#82
Define an equipment system where players can update their character's equipment automatically after the purchase.

Commit:
#82 - Remove boiler plate code to auto equip purchase items bought directly from the store
If an equipment item is "drop", it'll move back to the inventory
